### ISSUE TITLE
Support custom loss function for model training and evaluation

### ISSUE THREAD

**Issue by caisq** [tensorflow/tfjs#198]
To get help from the community, check out our [Google group](https://groups.google.com/a/tensorflow.org/forum/#!forum/tfjs).

#### TensorFlow.js version
0.9.1

Currently, if one tries to define a custom loss function and use it in training of evaluation of a tf.Model, a la Python Keras, an error will be thrown saying unknown loss. losses.get() needs to be revised to fix that.
